src/game.rs

```diff
@@ -0,0 +1,250 @@
+use macroquad::{
+    color::{BLACK, WHITE},
+    shapes::draw_rectangle,
+    window::{clear_background, next_frame},
+};
+
+/// Represents the state of the Game of Life.
+struct Game {
+    height: usize,
+    width: usize,
+    cells: Vec<bool>,
+}
+
+impl Game {
+    /// Creates a new Game of Life with random initial state.
+    ///
+    /// # Arguments
+    ///
+    /// * `height` - The height of the game grid.
+    /// * `width` - The width of the game grid.
+    ///
+    /// # Returns
+    ///
+    /// A new `Game` instance.
+    fn new(height: usize, width: usize) -> Self {
+        let cells = (0..height * width).map(|_| rand::random()).collect();
```

Comment:
aren't you creating a cell for every pixel here? shouldn't you divide the grid size by the cell size?

Reply:
Yeah sure, but I what I meant is: aren't you creating more cells than needed?

Reply:
For instance, if the screen is 1024x800 -> size(cells) = 819.200. Now if your cells size is, say, 5, then size(cells) = 1024/5 * 800/5 = 32.768

Reply:
I'm not sure honestly. Maybe if there are more cells than they fit in the screen there could be some issues with the way the program draws to the screen now that i think about it.

Reply:
> For instance, if the screen is 1024x800 -> size(cells) = 819.200. Now if your cells size is, say, 5, then size(cells) = 1024/5 * 800/5 = 32.768

I don't get this, the height and width passed to the `new` function are not necessarily the screen height and width. They are hard coded in main.rs in this case.

Reply:
Imagine you pass the screen size as params, but it applies to any value

Reply:
Yeah, then it does get messed up i just tried it. But if you pass sreen-size/cell-size as params it works alright

Reply:
Didn't notice that, I'll fix it.